File: 3079. Find the Sum of Encrypted Integers.py
class Solution:
    def sumOfEncryptedInt(self, nums: List[int]) -> int:
        sum = 0
        for i in nums:
            sum += self.encrypt(i)
        return sum
    def encrypt(self, i):
        max_digit = 0
        copy = i
        while copy > 0:
            max_digit = max(max_digit, copy%10)
            copy //= 10
        answer = 0
        while i > 0:
            answer *= 10
            answer += max_digit
            i //= 10
        return answer

File: 3079. Find the Sum of Encrypted Integers.java
class Solution {
    public int sumOfEncryptedInt(int[] nums) {
        int sum = 0;
        for (int i : nums) sum += encrypt(i);
        return sum;
    }
    private int encrypt(int i) {
        int maxDigit = 0;
        for (char c : String.valueOf(i).toCharArray()) maxDigit = Math.max(maxDigit, c - '0');
        int answer = 0;
        while (i > 0) {
            answer *= 10;
            answer += maxDigit;
            i /= 10;
        }
        return answer;
    }
}

